Storage/src/Connection/Rest.php

```diff
@@ -304,6 +322,10 @@ public function insertObject(array $args = [])
             ]
         ];
 
+        if (isset($args['ifGenerationMatch'])) {
```

Comment:
Probably I should have mentioned explicitly, but we also need to add the same for `ifGenerationNotMatch` and since that's the case, probably we should refactor this to take these preconditions from a list/map and then act accordingly.

Reply:
Would we need to add Metageneration preconditions also?

cc: @ajupazhamayil

Reply:
Not for uploads. In this we only need the `ifGenerationNotMatch` precondition.

Reply:
Okay, Implemented in commit [e8174b5](https://github.com/googleapis/google-cloud-php/pull/5730/commits/e8174b581f2af6ee61c7822010d426e46fc92a80)